testing/internal/e2e/scope/scope_test.go

```diff
@@ -0,0 +1,322 @@
+		"-scope-id", scopeReply.Item.Id,
+		"-format", "json",
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+	err = json.Unmarshal(output.Stdout, &jobs)
+	require.NoError(t, err)
+	assert.Len(t, jobs.Items, 0)
+
+	output = e2e.RunCommand(
+		"boundary", "scopes", "destroy-key-version",
+		"-scope-id", scopeReply.Item.Id,
+		"-key-version-id", databaseKeyVersionToDestroy.Id,
+		"-format", "json",
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+	err = json.Unmarshal(output.Stdout, &destruction)
+	require.NoError(t, err)
+	// Most data key versions should be destroyed asynchronously, as they require re-encryption
+	// of existing data.
+	assert.Equal(t, "pending", destruction.Item.State)
+
+	// The default scheduler job run interval is 1 minute. The runs become available 1
+	// second after the last successful run. We need to re-encrypt data in 1 table,
+	// and then remove the key. This job should take between 1 and 2 minutes to run,
+	// depending on the timing of the first started run.
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+	t.Cleanup(cancel)
+	for {
+		output = e2e.RunCommand(
+			"boundary", "scopes", "list-key-version-destruction-jobs",
+			"-scope-id", scopeReply.Item.Id,
+			"-format", "json",
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+		err = json.Unmarshal(output.Stdout, &jobs)
+		require.NoError(t, err)
+		if len(jobs.Items) == 0 {
+			break
+		}
+		require.Len(t, jobs.Items, 1)
+		assert.Equal(t, databaseKeyVersionToDestroy.Id, jobs.Items[0].KeyVersionId)
+		select {
+		case <-time.After(5 * time.Second):
+		case <-ctx.Done():
+			t.Fatal("Test timed out waiting for destruction to finish")
+		}
+	}
+
+	output = e2e.RunCommand(
+		"boundary", "scopes", "list-keys",
+		"-scope-id", scopeReply.Item.Id,
+		"-format", "json",
+	)
+	require.NoError(t, output.Err, string(output.Stderr))
+	err = json.Unmarshal(output.Stdout, &keys)
+	require.NoError(t, err)
+	assert.Len(t, keys.Items, 7)
+	for _, key := range keys.Items {
+		switch key.Purpose {
+		case "rootKey", "database":
+			assert.Len(t, key.Versions, 1)
+		default:
+			assert.Len(t, key.Versions, 2)
+		}
+	}
+
+	t.Logf("Successfully destroyed a root key and data key")
+}
+
+// TestKeyDestructionApi uses the boundary CLI to test key destruction.
+func TestKeyDestructionApi(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+
+	client, err := boundary.NewApiClient()
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	sc := scopes.NewClient(client)
+
+	scope, err := sc.Create(ctx, "global", scopes.WithName("testscope"))
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		_, err := sc.Delete(context.Background(), scope.Item.Id)
+		require.NoError(t, err)
+	})
+
+	keys, err := sc.ListKeys(ctx, scope.Item.Id)
+	require.NoError(t, err)
+	assert.Len(t, keys.Items, 7)
+	// Assert that all keys have the same number of versions
+	numVersions := len(keys.Items[0].Versions)
```

Comment:
Just pointing out that this was done a little differently in the cli test (it was hard-coding numbers). I assume now that the tests are in different scopes, you can still assume there should only be 1 version here?

Reply:
Good shout, I missed changing this back